### DISCUSSION TITLE
Build errored out - 404 Not found

### DISCUSSION THREAD

**Discussion by Sfekke** [vmware/packer-examples-for-vsphere#144] [Help] [answered]
Hey all,

I've been trying to build WS2022 but get hung up pretty early in the process.
All configuration seems correct yet I must be missing something as it 404's

A log might be in order : 
(There's more included than perhaps necessary, at any rate feel free to skip right to the last line)

```
==> vsphere-iso.windows-server-standard-core: Pausing after run of step 'StepConnect'. Press enter to continue.
2021/12/13 11:53:01 packer-plugin-vsphere_v1.0.2_x5.0_linux_amd64 plugin: 2021/12/13 11:53:01 No URLs were provided to Step Download. Continuing...
==> vsphere-iso.windows-server-standard-core: Pausing after run of step 'StepDownload'. Press enter to continue.
==> vsphere-iso.windows-server-standard-core: Creating CD disk...
2021/12/13 11:53:02 packer-plugin-vsphere_v1.0.2_x5.0_linux_amd64 plugin: 2021/12/13 11:53:02 CD path: /tmp/packer1141371933.iso
2021/12/13 11:53:02 packer-plugin-vsphere_v1.0.2_x5.0_linux_amd64 plugin: 2021/12/13 11:53:02 Wrote 2019 bytes to /tmp/packer_to_cdrom3872691847/windows-init.ps1
2021/12/13 11:53:02 packer-plugin-vsphere_v1.0.2_x5.0_linux_amd64 plugin: 2021/12/13 11:53:02 Wrote 6376 bytes to /tmp/packer_to_cdrom3872691847/windows-prepare.ps1
2021/12/13 11:53:02 packer-plugin-vsphere_v1.0.2_x5.0_linux_amd64 plugin: 2021/12/13 11:53:02 Wrote 2026 bytes to /tmp/packer_to_cdrom3872691847/windows-server-init.ps1
2021/12/13 11:53:02 packer-plugin-vsphere_v1.0.2_x5.0_linux_amd64 plugin: 2021/12/13 11:53:02 Wrote 6383 bytes to /tmp/packer_to_cdrom3872691847/windows-server-prepare.ps1
2021/12/13 11:53:02 packer-plugin-vsphere_v1.0.2_x5.0_linux_amd64 plugin: 2021/12/13 11:53:02 Wrote 4285 bytes to /tmp/packer_to_cdrom3872691847/windows-vmtools.ps1
2021/12/13 11:53:02 packer-plugin-vsphere_v1.0.2_x5.0_linux_amd64 plugin: 2021/12/13 11:53:02 Wrote 1260 bytes to /tmp/packer_to_cdrom3872691847/root-ca.cer
2021/12/13 11:53:02 packer-plugin-vsphere_v1.0.2_x5.0_linux_amd64 plugin: 2021/12/13 11:53:02 Executing: /usr/bin/xorriso [-as genisoimage -rock -joliet -volid packer -output /tmp/packer1141371933.iso /tmp/packer_to_cdrom3872691847]
    vsphere-iso.windows-server-standard-core: xorriso 1.4.8 : RockRidge filesystem manipulator, libburnia project.
    vsphere-iso.windows-server-standard-core: Drive current: -outdev 'stdio:/tmp/packer1141371933.iso'
    vsphere-iso.windows-server-standard-core: Media current: stdio file, overwriteable
    vsphere-iso.windows-server-standard-core: Media status : is blank
    vsphere-iso.windows-server-standard-core: Media summary: 0 sessions, 0 data blocks, 0 data, 6214m free
    vsphere-iso.windows-server-standard-core: xorriso : WARNING : -volid text does not comply to ISO 9660 / ECMA 119 rules
    vsphere-iso.windows-server-standard-core: Added to ISO image: directory '/'='/tmp/packer_to_cdrom3872691847'
    vsphere-iso.windows-server-standard-core: xorriso : UPDATE : 7 files added in 1 seconds
    vsphere-iso.windows-server-standard-core: xorriso : UPDATE : 7 files added in 1 seconds
    vsphere-iso.windows-server-standard-core: ISO image produced: 203 sectors
    vsphere-iso.windows-server-standard-core: Written to medium : 203 sectors at LBA 0
    vsphere-iso.windows-server-standard-core: Writing to 'stdio:/tmp/packer1141371933.iso' completed successfully.
    vsphere-iso.windows-server-standard-core: Done copying paths from CD_dirs
==> vsphere-iso.windows-server-standard-core: Pausing after run of step 'StepCreateCD'. Press enter to continue.
==> vsphere-iso.windows-server-standard-core: Uploading packer1141371933.iso to packer_cache/packer1141371933.iso
==> vsphere-iso.windows-server-standard-core: Pausing before cleanup of step 'StepCreateCD'. Press enter to continue.
2021/12/13 11:53:03 packer-plugin-vsphere_v1.0.2_x5.0_linux_amd64 plugin: 2021/12/13 11:53:03 Deleting CD disk: /tmp/packer1141371933.iso
==> vsphere-iso.windows-server-standard-core: Pausing before cleanup of step 'StepDownload'. Press enter to continue.
==> vsphere-iso.windows-server-standard-core: Pausing before cleanup of step 'StepConnect'. Press enter to continue.
**Build 'vsphere-iso.windows-server-standard-core' errored after 9 seconds 165 milliseconds: 404 Not Found**
```

Am I dealing with a connection issue to our vCenter?
The account used has full administrative rights.

Thanks in advance!

- Sven

_Cleaned up the post & included proper formatting._

**Comment by tenthirtyam** [comment, marked as answer]
Share version of vSphere?

Ryan

**Reply by Sfekke** [reply]
> Share version of vSphere?
> 
> Ryan

Totally forgot to add this, my bad!

I'm using **vSphere 7.0.2.00500**

**Reply by tenthirtyam** [reply]
Can you provided any additional environmental details?

Ryan

**Reply by Sfekke** [reply]
Our vCenter environment exists out of 12 clusters, 46 hosts & 1084 VM's.
They are all structured as in the screenshot included : 


![Cap](https://user-images.githubusercontent.com/36232209/145956501-3e3c7ec1-cd8f-47ed-b7d1-8d093cc31a17.PNG)

This means we adhere to the following hierarchy : 
- Datacenter
- Cluster
- Host
- VM

Currently the Packer service account has Administrator permissions, this is done through the Global Permissions (Propagated)
It's been created as a local account on vSphere.


Are there any details that might still be of use? If so I'd be happy to provide them!

- Sven

Additional edit :

After some more digging I noticed a "packer_cache" folder is successfully created on the common datastore; this leads me to believe it can't find the ISO

In the variables for WS2022 I've defined it as such : 

```
// Removable Media Settings
iso_path           = "ISOs" <- this folder on our common datastore houses the .ISO file - tried including the full path as well
iso_file           = "windows.iso" <- renamed for simplicity, all lowercase
iso_checksum_type  = "none" *<- tried with & without a checksum, same result.*
iso_checksum_value = "blank"
```

**Reply by tenthirtyam** [reply]
If you look deeper into the example, you'll see the following:

```hcl
  // Removable Media Settings
  iso_paths    = ["[${var.common_iso_datastore}] ${var.iso_path}/${var.iso_file}", "[] /vmimages/tools-isoimages/${var.vm_guest_os_family}.iso"]
  iso_checksum = "${var.iso_checksum_type}:${var.iso_checksum_value}"
  cd_files = [
    "${path.cwd}/scripts/${var.vm_guest_os_family}/",
    "${path.cwd}/certificates/"
  ]
```

You'll need to ensure that that the hosts have the VMware Tools images and that `var.vm_guest_os_family` is "`windows"` as the VMware Tools iso is mounted as E:\ and is used by the generated autounattend.xml for the tools installation..

You'll also need to ensure that "`[${var.common_iso_datastore}] ${var.iso_path}/${var.iso_file}`" is your defined path to the ISO.

e.g. `[my_iso_datastore] my_iso_path/my_iso_file.iso`

I recommend adding back the `iso_checksum_type` and `iso_checksum_value` variables.

Ryan

**Reply by Sfekke** [reply]
VMware Tools was indeed not defined correctly on my end, thanks for clearing that up!
![Capture](https://user-images.githubusercontent.com/36232209/146164730-668f3d17-74c3-40b0-a889-5e6c900edafe.PNG)

I have not modified the HCL files _much_ so I'll presume it's mounting to E:\ by default.

Sadly however even after triple checking my path
`VMSVC-LENOVO-ISO-FILES ISOs/windows.iso`
Each part being defined seemingly correct as : Common datastore -> Path -> ISO File

Included below is the current common.pkrvars.hcl
```
// Template and Content Library Settings
common_template_conversion         = false
common_content_library_name        = "ISOs"
common_content_library_ovf         = true
common_content_library_destroy     = true
common_content_library_skip_import = false

// Removable Media Settings
common_iso_datastore = "VMSVC-LENOVO-ISO-FILES"
```

For testing's purpose I've placed the .ISO in the content library as well; despite my best efforts I'm yet again greeted by a 404
![Content Library](https://user-images.githubusercontent.com/36232209/146166271-3557491a-6308-41ff-afae-32af5a6c9b86.png)

Regardless of that I'd still like to thank you for helping me out so far!

**Reply by tenthirtyam** [reply]
There's currently no support for using an ISO from the content library; it must exist on a datastore.

Ryan

**Reply by Sfekke** [reply]
I didn't explicitly state this before but the .ISO is available on the datastore.
Using the content library was just a quick test to see if that'd resolve the 404.

![Storage](https://user-images.githubusercontent.com/36232209/146521759-fb82d99d-a675-4cd3-982b-32d438851552.png)

After trying to get things sorted there's still no progress, is there any more information I can provide that would help?

- Sven

**Reply by tenthirtyam** [reply]
I'm stepping away from the screen for the holidays until 1/3. If you'd like to do a zoom in the New Year I'd be happy to look at it with you. 

Let me know.

Ryan

**Reply by Sfekke** [reply]
First & foremost, happy holidays!
I'd like to go over things in a call at a time that works for you, via LinkedIn I've sent you a quick connection request.

We can swap emails there to setup a Zoom call.

Thanks in advance for taking the time out of your day to do so!

- Sven
